wdl-lint/src/rules/command_shellcheck.rs

```diff
@@ -0,0 +1,357 @@
+                        count_leading_whitespace(
+                            pholder_str.strip_prefix("\n").unwrap_or(pholder_str),
+                        )
+                    }
+                }
+            } else {
+                0
+            }
+        };
+
+        // Map each actual line of the command to its corresponding
+        // CommandPart and start position
+        let mut line_map = HashMap::new();
+        let mut on_same_line = false;
+        let mut line_num = 0usize;
+        for part in section.parts() {
+            match part {
+                CommandPart::Text(ref text) => {
+                    let mut last_line = 0usize;
+                    for (_, start, _) in lines_with_offset(text.as_str()) {
+                        // Check to see if we are still on the previous line
+                        // This happens after we encounter a placeholder
+                        if on_same_line {
+                            on_same_line = false;
+                            last_line = last_line.saturating_sub(1);
+                        }
+                        line_map.insert(line_num + last_line, (part.clone(), start));
+                        last_line += 1;
+                    }
+                    line_num += last_line;
+                }
+                CommandPart::Placeholder(_) => {
+                    on_same_line = true;
+                }
+            }
+        }
+
+        match run_shellcheck(&sanitized_command) {
+            Ok(comments) => {
+                for comment in comments {
+                    // skip declarations that shellcheck is unaware of
+                    if comment.code == 2154
+                        && decls.contains(comment.message.split_whitespace().next().unwrap_or(""))
+                    {
+                        continue;
+                    }
+                    let (rel_part, start) = line_map
+                        .get(&comment.line)
+                        .expect("shellcheck line corresponds to command line");
+                    let rel_token = match rel_part {
+                        CommandPart::Text(text) => text.syntax(),
+                        CommandPart::Placeholder(phold) => {
+                            &support::token(phold.syntax(), SyntaxKind::PlaceholderNode)
+                                .expect("should have a placeholder node token")
+                        }
+                    };
+                    let inner_span = {
+                        let outer_span = rel_token.text_range().to_span();
+                        Span::new(
+                            outer_span.start() + leading_whitespace + start + comment.column - 1,
+                            comment.end_column - comment.column,
+                        )
+                    };
+                    state.exceptable_add(
+                        shellcheck_lint(&comment, inner_span),
+                        SyntaxElement::from(rel_token.clone()),
```

Comment:
The `element` here can just be the `section`, as that's the lowest granularity we allow `except` comments at. As is, we're going to check a bunch of lower level ancestors that we know won't have a comment.